types/react/index.d.ts

```diff
@@ -517,7 +517,7 @@ declare namespace React {
     type FC<P = {}> = FunctionComponent<P>;
 
     interface FunctionComponent<P = {}> {
-        (props: P, context?: any): ReactElement<any, any> | null;
+        (props: PropsWithChildren<P>, context?: any): ReactElement<any, any> | null;
```

Comment:
The behavior without `PropWithChildren` is intended. Check https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210 for more information.